src/test/spec/unified_runner/mod.rs

```diff
@@ -0,0 +1,368 @@
+                    let client = self.entities.get(&database.client).unwrap().as_client();
+                    let database = if let Some(ref options) = database.database_options {
+                        let options = options.as_database_options();
+                        client.database_with_options(&database.database_name, options)
+                    } else {
+                        client.database(&database.database_name)
+                    };
+                    (id, database.into())
+                }
+                TestFileEntity::Collection(collection) => {
+                    let id = collection.id.clone();
+                    let database = self
+                        .entities
+                        .get(&collection.database)
+                        .unwrap()
+                        .as_database();
+                    let collection = if let Some(ref options) = collection.collection_options {
+                        let options = options.as_collection_options();
+                        database.collection_with_options(&collection.collection_name, options)
+                    } else {
+                        database.collection(&collection.collection_name)
+                    };
+                    (id, collection.into())
+                }
+                TestFileEntity::Session(_) => {
+                    panic!("Explicit sessions not implemented");
+                }
+                TestFileEntity::Bucket(_) => {
+                    panic!("GridFS not implemented");
+                }
+            };
+            if self.entities.insert(id.clone(), entity).is_some() {
+                panic!("Entity with id {} already present in entity map", id);
+            }
+        }
+    }
+
+    pub fn get_client(&self, id: &str) -> &ClientEntity {
+        self.entities.get(id).unwrap().as_client()
+    }
+
+    pub fn get_database(&self, id: &str) -> &Database {
+        self.entities.get(id).unwrap().as_database()
+    }
+
+    pub fn get_collection(&self, id: &str) -> &Collection {
+        self.entities.get(id).unwrap().as_collection()
+    }
+}
+
+pub async fn run_unified_format_test(test_file: TestFile) {
+    let version_matches = SPEC_VERSIONS.iter().any(|req| {
+        if req.major != test_file.schema_version.major {
+            return false;
+        }
+        if req.minor < test_file.schema_version.minor {
+            return false;
+        }
+        // patch versions do not affect the test file format
+        true
+    });
+    if !version_matches {
+        panic!(
+            "Test runner not compatible with specification version {}",
+            &test_file.schema_version
+        );
+    }
+
+    let mut test_runner = TestRunner::new().await;
+
+    if let Some(requirements) = test_file.run_on_requirements {
+        let mut can_run_on = false;
+        for requirement in requirements {
+            if requirement.can_run_on(&test_runner.internal_client).await {
```

Comment:
I previously had this using the iterator `any` method but changing `can_run_on` to be `async` broke it -- is there a better way to do this?

Reply:
The only way I can think of is to fold into a `Future<Output=bool>` and then `await`, but I think the way you have it here is cleaner.